### PR TITLE
Feature/iiwa 140

### PR DESCRIPTION
**Pull request by charkour**
Merge all hardware POST subtasks into the monitor detail page story.

### CODE REVIEW

File: src/backend-api/Controllers/AddController.cs [hunk 1]
```diff
@@ -514,6 +509,229 @@ public IActionResult PostProgram([FromBody] PostProgramInputModel input)
...
+         *              employeeName: string,
+         *              employeeId: int,
+         *             } ,.. ]
+         * Will return an array of objects with the employee name and
+         *   id for every non-archived employee to be used when assigning 
+         *   a piece of hardware.
+         */
+        [HttpGet]
+        [Route("HardwarePrep")]
+        public IActionResult GetMonitorPrep()
```
Comment (Xiscoder) [inline src/backend-api/Controllers/AddController.cs]: get hardware prep

File: src/backend-api/Controllers/AddController.cs [hunk 2]
```diff
@@ -514,6 +509,229 @@ public IActionResult PostProgram([FromBody] PostProgramInputModel input)
...
+            // Update values we don't want touched by the endpoint call.
+            hardware.IsAssigned = EmployeeId != null ? true : false;
+            hardware.IsDeleted = false;
+
+            // Get class name at runtime
+            string type = GetClassName(hardware);
+
+            try
+            {
+                table.Add(hardware);
```
Comment (Xiscoder) [inline src/backend-api/Controllers/AddController.cs:687]: How are we managing monthsPerRenewal controlling the isCostPerYear

File: src/backend-api/Controllers/AddController.cs [hunk 3]
```diff
@@ -514,6 +509,229 @@ public IActionResult PostProgram([FromBody] PostProgramInputModel input)
...
+
+            // If we make it here, everything must have succeeded
+            return StatusCode(201);
+        }
+
+        /* UpdateHardwareHistory(empId, hardwareType, hardwareId, eventType, date) will add an entry into the 
+         *   hardware history table.
+         */
+        // TODO: Abstract this
+        private void UpdateHardwareHistory(int? empId, string hardwareType, int hardwareId, string eventType, DateTime? date)
```
Comment (Xiscoder) [inline src/backend-api/Controllers/AddController.cs:726]: Consider pushing this up to the contextController

File: src/backend-api/Controllers/AddController.cs (hunk 2 above)
Comment (charkour) [reply src/backend-api/Controllers/AddController.cs:687]: Delegated to front-end

**Pull request merged**